### PR TITLE
Redirection to file should truncate when possible

### PR DESCRIPTION
**Pull request by tiago4orion**
The redirection tests sucks.. I'll improve that

Closes #216

### CODE REVIEW

File: internal/sh/shell.go [hunk 1]
```diff
@@ -1106,7 +1105,7 @@ func (shell *Shell) openRedirectLocation(location ast.Expr) (io.WriteCloser, err
 	}
 
 	if protocol == "" {
-		return os.OpenFile(locationStr, os.O_RDWR|os.O_CREATE, 0644)
+		return os.OpenFile(locationStr, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
```
Comment (katcipis) [inline internal/sh/shell.go]: os.Create has different flags ?

File: internal/sh/shell_test.go [hunk 2]
```diff
@@ -546,30 +546,40 @@ func TestExecuteRedirection(t *testing.T) {
...
+		t.Fatal(err)
+	}
 
+	if string(content) != "hello world" {
+		t.Fatalf("File differ: '%s' != '%s'", string(content), "hello world")
+	}
+
+	// Test redirection truncate the file
+	err = shell.Exec("redirect", `
+        echo -n "a" > `+path+`
```
Comment (katcipis) [inline internal/sh/shell_test.go]: could use fmt.Sprintf to generate the code, don't know what is worse although =)

File: internal/sh/shell.go (hunk 1 above)
Comment (tiago4orion) [inline internal/sh/shell.go]: same flags, but different permissions

Comment (katcipis) [inline internal/sh/shell.go]: I see =)

**Comment by tiago4orion** [comment]
@katcipis @vitorarins Can we merge this also?

**Pull request merged**